### ISSUE TITLE
Align NukeUI.Image self-sizing behavior with SwiftUI.Image

### ISSUE THREAD

**Issue by robmaceachern** [kean/Nuke#578]
I've noticed that there are some layout differences between `SwiftUI.Image` and `NukeUI.Image` when using `maxWidth/maxHeight` in a `List`. I'm not sure if it's intentional or not, but it definitely caught me by surprise when I was comparing `AsyncImage` and `LazyImage` behaviour.

I've also noticed that the `NukeUI.Image` can cause sibling `Text` to truncate after a containing `HStack` appears after being scrolled offscreen. Sample code and screenshots below. The image used was downloaded from `https://picsum.photos/id/0/5616/3744`

Xcode: 13.4.1
iOS 15.5

```swift
import SwiftUI
import NukeUI

struct ContentView: View {
    var body: some View {
        List {
            HStack {
                NukeUI.Image(UIImage(named: "laptop")!)
                    .resizingMode(.fill)
                    .frame(width: 100, height: 100)
                Text("Nuke - frame(width: 100, height: 100):")
            }

            HStack {
                SwiftUI.Image(uiImage: UIImage(named: "laptop")!)
                    .resizable()
                    .frame(width: 100, height: 100)
                Text("SwiftUI - frame(width: 100, height: 100):")
            }

            HStack {
                NukeUI.Image(UIImage(named: "laptop")!)
                    .resizingMode(.fill)
                    .frame(maxWidth: 100, maxHeight: 100)
                Text("Nuke - frame(maxWidth: 100, maxHeight: 100):")
            }

            HStack {
                SwiftUI.Image(uiImage: UIImage(named: "laptop")!)
                    .resizable()
                    .frame(maxWidth: 100, maxHeight: 100)
                Text("SwiftUI - frame(maxWidth: 100, maxHeight: 100)")
            }

            HStack {
                Rectangle()
                    .frame(width: 100, height: 800)
                Text("A long rectangle")
            }
        }
    }
}
```

<img width="506" alt="CleanShot 2022-07-27 at 11 01 21@2x" src="https://user-images.githubusercontent.com/430436/181294650-70e13771-3612-41dc-83ab-9927d416ae74.png">

![CleanShot 2022-07-27 at 11 08 26@2x](https://user-images.githubusercontent.com/430436/181296245-621b2d4e-a763-4c84-a013-37279128eb4b.png)

**Comment by kean** [comment]
Hi, @robmaceachern. It wasn't designed to work the same way as `AsyncImage` in terms of layout, and it's most likely impossible given the limitations of the API.  The only supported use case is when both dimensions are set explicitly.

There is a new [`sizeThatFits`](https://developer.apple.com/documentation/uikit/uiview/1622625-sizethatfits) API on `UIViewRepresentable` that could probably be used to align the behavior with `AsyncImage`.

**Comment by kean** [comment]
Addressed in [Nuke 12.0 (Beta 1)](https://github.com/kean/Nuke/releases/tag/12.0.0-beta.1)

**Comment by kean** [comment]
Please test it in [Nuke 12 (RC1)](https://github.com/kean/Nuke/releases/tag/12.0.0-rc.1).
